force-app/main/default/lwc/recordPickerDynamicTarget/recordPickerDynamicTarget.js

```diff
@@ -0,0 +1,82 @@
+import { wire, LightningElement, api } from 'lwc';
+import { getObjectInfos } from 'lightning/uiObjectInfoApi';
+
+export default class RecordPickerDynamicTarget extends LightningElement {
+    placeholder = 'Search...';
+
+    @api
+    label = 'Select a record';
+
+    objectApiNames = ['Account', 'Case', 'Contact'];
+    selectedTarget = 'Account';
+    objectInfos = [];
+    isObjectInfoLoading = true;
+    currentSelectedRecordId = null;
+
+    displayInfos = {
+        Account: {
+            additionalFields: ['Type']
+        },
+        Contact: {
+            additionalFields: ['Phone']
+        }
+    };
+
+    matchingInfos = {
+        Account: {
+            additionalFields: [{ fieldPath: 'Type' }]
+        },
+        Contact: {
+            additionalFields: [{ fieldPath: 'Phone' }]
+        }
+    };
+
+    get displayInfo() {
+        return this.displayInfos?.[this.selectedTarget];
+    }
+
+    get matchingInfo() {
+        return this.matchingInfos?.[this.selectedTarget];
+    }
+
+    get targetObjects() {
+        if (this.isObjectInfoLoading) {
+            return [];
+        }
+
+        return this.objectInfos.map((objectInfo) => {
+            return {
+                label: objectInfo.label,
+                value: objectInfo.apiName
+            };
+        });
+    }
+
+    get showTargetSelector() {
+        return this.currentSelectedRecordId === null;
+    }
+
+    handleTargetSelection(event) {
+        this.selectedTarget = event.target.value;
+        this.refs.recordPicker.clearSelection();
+    }
+
+    handleRecordSelect(event) {
+        this.currentSelectedRecordId = event.detail.recordId;
+    }
+
+    @wire(getObjectInfos, { objectApiNames: '$objectApiNames' })
+    wiredGetObjectInfos({ error, data }) {
+        if (error) {
+            this.dispatchEvent(new CustomEvent('error', { error: error }));
```

Comment:
Since we're building a standalone component, let's display the error within the component.
Feel free to reuse the error handling logic from `wireGetRecord`.